crates/bevy_ecs/src/system/function_system.rs

```diff
@@ -528,6 +530,17 @@ where
     }
 }
 
+/// SAFETY: `F`'s param is `ReadOnlySystemParam`, so this system will only read from the world.
+unsafe impl<In, Out, Param, Marker, F> ReadOnlySystem for FunctionSystem<In, Out, Param, Marker, F>
```

Comment:
Does `ReadOnlySystem` need to be implemented for `ExclusiveFunctionSystem` as well?

Reply:
Exclusive systems have access to `&mut World`, so they are able to modify the world.

Reply:
I was thinking of a system that uses `&World` but I guess that's a `FunctionSystem`